pkgs/games/anki/default.nix

```diff
@@ -1,199 +1,265 @@
+    inherit version src yarnOfflineCache;
+
+    nativeBuildInputs = [
+      fixup_yarn_lock
+      yarn
+      nodejs-slim
+    ];
+
+    configurePhase = ''
+      export HOME=$NIX_BUILD_TOP
+      yarn config --offline set yarn-offline-mirror $yarnOfflineCache
+      fixup_yarn_lock yarn.lock
+      yarn install --offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules/
+      yarn run postinstall --offline
+    '';
+
+    installPhase = ''
+      mv node_modules $out
+    '';
+  };
+in
+python39.pkgs.buildPythonApplication {
+  inherit pname version src;
+
   outputs = [ "out" "doc" "man" ];
 
-  propagatedBuildInputs = [
-    pyqtwebengine
-    sqlalchemy
+  patches = [
+    ./gl-fixup.patch
+    ./no-update-check.patch
+    ./0001-Don-t-download-nodejs-if-NODE_BINARY-is-set.patch
+    ./0002-Allow-setting-YARN_BINARY-for-the-build-system.patch
+  ];
+
+  inherit cargoDeps;
+
+  nativeBuildInputs = [
+    fakeGit
+    fixup_yarn_lock
+    offlineYarn
+
+    rustPlatform.rust.cargo
+    rustPlatform.cargoSetupHook
+    ninja
+    qt6.wrapQtAppsHook
+    rsync
+  ];
+  nativeCheckInputs = with python39.pkgs; [ pytest mock astroid  ];
+
+  buildInputs = [
+    qt6.qtbase
+  ];
+  propagatedBuildInputs = with python39.pkgs; [
+    # This rather long list came from running:
+    #    grep --no-filename -oE "^[^ =]*" python/{requirements.base.txt,requirements.bundle.txt,requirements.qt6_4.txt} | \
+    #      sort | uniq | grep -v "^#$"
+    # in their repo at the git tag for this version
+    # There's probably a more elegant way, but the above extracted all the
+    # names, without version numbers, of their python dependencies. The hope is
+    # that nixpkgs versions are "close enough"
+    # I then removed the ones the check phase failed on (pythonCatchConflictsPhase)
     beautifulsoup4
-    send2trash
-    pyaudio
-    requests
+    certifi
+    charset-normalizer
+    click
+    colorama
     decorator
-    markdown
+    distro
+    flask
+    flask-cors
+    idna
+    importlib-metadata
+    itsdangerous
+    jinja2
     jsonschema
-    setuptools
-  ]
-  ++ lib.optional plotsSupport matplotlib
-  ++ lib.optionals stdenv.isDarwin [ CoreAudio ]
-  ;
+    markdown
+    markupsafe
+    orjson
+    pep517
+    python39.pkgs.protobuf
+    pyparsing
+    pyqt6
+    pyqt6-sip
+    pyqt6-webengine
+    pyrsistent
+    pysocks
+    requests
+    send2trash
+    six
+    soupsieve
+    urllib3
+    waitress
+    werkzeug
+    zipp
+  ] ++ lib.optionals stdenv.isDarwin [ CoreAudio ];
 
-  nativeCheckInputs = [ pytest glibcLocales mock nose ];
+  # Activate optimizations
+  RELEASE = true;
 
-  nativeBuildInputs = [ pyqtwebengine.wrapQtAppsHook ];
-  buildInputs = [ lame mpv-unwrapped libpulseaudio ];
+  PROTOC_BINARY = "${protobuf}/bin/protoc";
+  NODE_BINARY = "${nodejs}/bin/node";
+  YARN_BINARY = "${offlineYarn}/bin/yarn";
+  PYTHON_BINARY = "${python39}/bin/python";
```

Comment:
Nit: You can probably utilise `lib.getExe` here. I only recently learned about it too and it's pretty handy.

Reply:
Neat, I didn't know about that either.

```
$ rg 'lib.getExe' | wc -l
29
$ rg --pcre2 '\$\{([^}]+)\}/bin/\1( |")' | wc -l
1050
```

It doesn't seem very popular though.
I don't mind switching though, sure, done. Thanks for teaching me about that!